packages/graphql/lib/koaMiddleware.ts

```diff
@@ -0,0 +1,130 @@
+import { GraphQLSchema, GraphQLError, GraphQLFormattedError } from "graphql";
+import { ApolloServer, gql, Config, ApolloError, UserInputError, AuthenticationError, ForbiddenError } from "apollo-server-koa";
+import * as _ from "lodash";
+
+import IGraphQlConfig from "./IGraphQlConfig";
+import { ILogger } from "@fullstack-one/logger";
+import { Koa } from "@fullstack-one/server";
+
+export function applyApolloMiddleware(app: Koa, schema: GraphQLSchema, config: IGraphQlConfig, logger: ILogger) {
```

Comment:
I saw that this function is only called ones in `packages/graphql/lib/index.ts`. However, this function could be called multiple times and would create multiple ApolloServers. Would it make sense to have a Singlton here? Something like:

```ts
let server;
export function applyApolloMiddleware(app: Koa, ...) {
  if (server != null) return;
  server = createApolloServer(...);
  ...
}
```

Reply:
Hmm, we could also create multiple Graphql endpoints in one server.

We are not doing it now, however, maybe later.